PWGGA/GammaConv/AliConversionMesonCuts.cxx

```diff
@@ -1349,16 +1349,16 @@ Bool_t AliConversionMesonCuts::MesonIsSelectedMCChiCAODESD(AliDalitzAODESDMC *fM
         // Select only -> ChiC radiative (JPsi+gamma) decay channel
         if(fMCMother->GetNDaughtersG()!=2)return kFALSE;
 
-            AliDalitzAODESDMC *jpsi   = 0x0;
-            AliDalitzAODESDMC *gamma   = 0x0;
-            AliDalitzAODESDMC *positron = 0x0;
-            AliDalitzAODESDMC *electron = 0x0;
+            auto_ptr<AliDalitzAODESDMC> jpsi   ;
+            auto_ptr<AliDalitzAODESDMC> gamma  ;
+            auto_ptr<AliDalitzAODESDMC> positron ;
+            auto_ptr<AliDalitzAODESDMC> electron ;
 
         //Int_t labeljpsiChiC = -1;
 
         for(Int_t index= fMCMother->GetFirstDaughterG();index<= fMCMother->GetLastDaughterG();index++){
             if(index < 0) continue;
-            AliDalitzAODESDMC* temp = mcEvent->Particle(index);
+            auto_ptr<AliDalitzAODESDMC> temp = mcEvent->Particle(index);
             //TParticle* temp = (TParticle*)mcEvent->Particle( index );
 
             switch( temp->GetPdgCodeG() ) {
```

Comment:
Here one of the big problems using unique_ptr, after defining the AliDalitzAODESDMC track for the temp, and using the switch, you can not copy a unique_ptr one to another very easily
`    //std::unique_ptr<AliDalitzAODESDMC> positron;
   // std::unique_ptr<AliDalitzAODESDMC> electron;
   // std::unique_ptr<AliDalitzAODESDMC> gamma;
    for(Int_t index= fMCMother->GetFirstDaughterG();index<= fMCMother->GetLastDaughterG();index++){
        if(index < 0) continue;
        std::shared_ptr<AliDalitzAODESDMC> temp = mcEvent->Particle(index);
        //TParticle* temp = (TParticle*)mcEvent->Particle( index );
        switch( temp->GetPdgCodeG() ) {
        case ::kPositron:
            std::shared_ptr<AliDalitzAODESDMC> positron = std::make_shared<AliDalitzAODESDMC>(*temp);
          //  auto positron      =  std::make_shared(temp);
            labelpositron = index;
            break;
        case ::kElectron:
          //  electron      =  temp;
            std::shared_ptr<AliDalitzAODESDMC> electron = std::make_shared<AliDalitzAODESDMC>(*temp);
         //auto electron      =  std::move(temp);
            labelelectron = index;
            break;`
One can only change one case at the time because you erase the temp previously with the "move" command, this is the only way you can completely pass the inputs to another unique_ptr, so the changes will rebuild all thinks of classes that share something and pass or copy some information, with respect the GetPdgCode function to lookout, the problem is the input you will give it to search for the info, and again you need something to define?

Reply:
Use three backticks  (```) so the code is readable:

\```
`code goes here`
\```

But yeah, the difficulty of copying is a feature, keeping the code as efficient as possible by avoiding accidental copying the whole object the pointer is pointing to.

In this case you're not creating any new `TParticles`, I don't see any need to change this code to use `std::unique_ptr` or `std::shared_ptr`.

Reply:
Dear Andrew,
yes on here, when I summon a temp 
`std::shared_ptr<AliDalitzAODESDMC>` temp = mcEvent->Particle(index);`
I wish to use on some case, and if is the type of particle I want, I will save all the TParticle information, the problem that this gives me is, one the way you can transfer from one `unique_ptr` to another `unique_ptr`,  after passing the first case,
```
        case ::kPositron:
            std::shared_ptr<AliDalitzAODESDMC> positron = std::make_shared<AliDalitzAODESDMC>(*temp);
          //  auto positron      =  std::make_shared(temp);
            labelpositron = index;
            break;
```
On the second case, I can use `temp` to compare on the next case.